### PR TITLE
Add note about `throwHttpErrors` and retry functionality

### PR DESCRIPTION
**Pull request by robertgr991**

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -303,6 +303,8 @@ Throw an `HTTPError` when, after following redirects, the response has a non-2xx
 
 Setting this to `false` may be useful if you are checking for resource availability and are expecting error responses.
 
+Note: Setting this to `false` will skip the retry functionality for non-2xx status codes because the response is considered successful, therefore no need to retry.
```
Comment (szmarczak) [inline readme.md:306]:
```suggestion
Note: If `false`, error responses are considered successful, therefore no retry will have place.
```

**Comment by sindresorhus** [comment]
You also need to update the docs in the TS doc comments for the option.

File: readme.md [hunk 2]
```diff
@@ -303,6 +303,8 @@ Throw an `HTTPError` when, after following redirects, the response has a non-2xx
 
 Setting this to `false` may be useful if you are checking for resource availability and are expecting error responses.
 
+Note: If `false`, error responses are considered successful, therefore no retry will have place.
```
Comment (sindresorhus) [inline readme.md:306]:
```suggestion
Note: If `false`, error responses are considered successful and the request will not be retried.
```

**Comment by sindresorhus** [comment]
> You also need to update the docs in the TS doc comments for the option.

**Comment by robertgr991** [comment]
Done.

**Pull request merged**